### PR TITLE
Remove `ID` from human and serialized fields

### PR DESCRIPTION
**Pull request by tmalikconfluent**
Release Notes
-------------

Breaking Changes
- 'id' has been removed from most of fields containing 'id' in their name

Checklist
---------
- [x] Leave this box unchecked if features are not yet available in production

What
----
Remove `ID` from human and serialized fields

References
----------
https://confluentinc.atlassian.net/browse/CLI-3014

Test & Review
-------------
Updated integration tests.

### CODE REVIEW

File: internal/kafka/command_cluster_describe.go [hunk 1]
```diff
@@ -35,8 +35,8 @@ type describeStruct struct {
 	Network            string `human:"Network,omitempty" serialized:"network,omitempty"`
 	Status             string `human:"Status" serialized:"status"`
 	Endpoint           string `human:"Endpoint" serialized:"endpoint"`
-	ByokKeyId          string `human:"BYOK Key ID" serialized:"byok_key_id"`
-	EncryptionKeyId    string `human:"Encryption Key ID" serialized:"encryption_key_id"`
```
Comment (brianstrauch) [inline internal/kafka/command_cluster_describe.go:39]: These are both external resources (outside of Confluent). There's a big difference between the "BYOK key" and "BYOK key ID" and both are strings. We might want to re-evaluate this change...

Comment (tmalikconfluent) [reply internal/kafka/command_cluster_describe.go:39]: okay, let me revert this one back. Any others that need to still contain ID?

File: internal/flink/command_artifact.go [hunk 2]
```diff
@@ -35,8 +35,8 @@ func printTable(cmd *cobra.Command, plugin connectcustompluginv1.ConnectV1Custom
 	table := output.NewTable(cmd)
 	table.Add(&flinkArtifactOut{
 		Name:          plugin.GetDisplayName(),
-		PluginId:      plugin.GetId(),
-		VersionId:     plugin.GetConnectorClass(),
+		Plugin:        plugin.GetId(),
+		Version:       plugin.GetConnectorClass(),
```
Comment (brianstrauch) [inline internal/flink/command_artifact.go:39]: @byang-confluent, are you ok with the CLI team making this change to improve standardization across our commands?

I'm also curious why "connector class" maps to "version ID". These seem like two separate things.

File: internal/kafka/command_cluster_describe.go (hunk 1 above)
Comment (brianstrauch) [reply internal/kafka/command_cluster_describe.go]: A general rule of thumb would be anything containing "Azure", "Aws", or "Gcp" should keep the "ID". For instance, `AzurePrivateLinkServiceId` should stay as is

Comment (brianstrauch) [reply internal/kafka/command_cluster_describe.go]: or `AwsVpcEndpoint`

File: internal/plugin/command_list.go [hunk 3]
```diff
@@ -16,7 +16,7 @@ import (
 
 type out struct {
 	PluginName string `human:"Name" serialized:"plugin_name"`
```
Comment (brianstrauch) [inline internal/plugin/command_list.go:18]:
```suggestion
	Name string `human:"Name" serialized:"name"`
```

File: internal/plugin/command_list.go [hunk 4]
```diff
@@ -16,7 +16,7 @@ import (
...
-	PluginId   string `human:"ID" serialized:"plugin_id"`
+	PluginId   string `human:"ID" serialized:"id"`
```
Comment (brianstrauch) [inline internal/plugin/command_list.go:19]:
```suggestion
	Id   string `human:"ID" serialized:"id"`
```

File: internal/flink/command_artifact.go (hunk 2 above)
Comment (byang-confluent) [reply internal/flink/command_artifact.go:39]: this is a temporary workaround that versioning api is not GA yet, but flink udf requires this information. 

we are working on versioning api this quarter to deliver this. and update cli after that. please keep this setting for now.

Comment (tmalikconfluent) [reply internal/flink/command_artifact.go:39]: reverting this back then

File: internal/kafka/command_cluster_describe.go (hunk 1 above)
Comment (tmalikconfluent) [reply internal/kafka/command_cluster_describe.go]: got it , reverting these ones

File: internal/flink/command_artifact.go (hunk 2 above)
Comment (brianstrauch) [reply internal/flink/command_artifact.go:39]: > please keep this setting for now.

We're not deleting it, we're just asking if we can change the field names from "Plugin ID" to "Plugin" and "Version ID" to "Version".

Comment (byang-confluent) [reply internal/flink/command_artifact.go:39]: yes, that should OK

**Review by byang-confluent — APPROVED** [review]

**Review by brianstrauch — CHANGES_REQUESTED** [review]
It doesn't look like these changes match the PR title

**Pull request closed without merging**